File: codeforces/contests/Codeforces Round 1013 (Div. 3)/E.py
import sys
import math


def sieve(limit):
    is_prime = [True] * (limit + 1)
    is_prime[0] = is_prime[1] = False
    for i in range(2, int(math.sqrt(limit)) + 1):
        if is_prime[i]:
            for j in range(i * i, limit + 1, i):
                is_prime[j] = False
    return is_prime


def count_interesting_ratios(n, is_prime):
    count = 0
    for a in range(1, n):
        for b in range(a + 1, n + 1):
            gcd_ab = math.gcd(a, b)
            F_ab = (a * b) // (gcd_ab * gcd_ab)
            if is_prime[F_ab]:
                count += 1
    return count


def main():
    input = sys.stdin.read
    data = input().split()
    t = int(data[0])
    cases = list(map(int, data[1:]))
    max_n = max(cases)

    is_prime = sieve(10**7)

    results = []
    for n in cases:
        results.append(str(count_interesting_ratios(n, is_prime)))

    sys.stdout.write("\n".join(results) + "\n")


if __name__ == "__main__":
    main()

File: codeforces/contests/Codeforces Round 1013 (Div. 3)/E.cpp
#include <bits/stdc++.h>
using namespace std;

const int MAX_N = 10000007;
vector<bool> is_prime(MAX_N, true);
vector<int> primes;

void sieve() {
    is_prime[0] = is_prime[1] = false;
    for (int i = 2; i * i < MAX_N; i++) {
        if (is_prime[i]) {
            for (int j = i * i; j < MAX_N; j += i) {
                is_prime[j] = false;
            }
        }
    }
    for (int i = 2; i < MAX_N; i++) {
        if (is_prime[i]) {
            primes.push_back(i);
        }
    }
}

int main() {
    sieve();
    int t;
    scanf("%d", &t);
    while (t--) {
        int n;
        scanf("%d", &n);
        long long ans = 0;
        for (int p : primes) {
            if (p > n) break;
            ans += (n / p);
        }
        printf("%lld\n", ans);
    }
    return 0;
}
